pkg/client/cli/cmd_gatherlogs.go

```diff
@@ -469,37 +468,29 @@ func anonymizeLog(stdout io.Writer, logFile string, anonymizer *anonymizer) erro
 // to anonymize.  It currently works for pods owned by StatefulSets,
 // ReplicaSets, and Deployments.
 func getSignificantPodNames(podName string) []string {
-	subNames := make([]string, 2)
-	nameComponents := strings.Split(podName, "-")
-	significantComponent := false
-	for i, component := range nameComponents {
-		// Indicates we found a Pod that comes from either a
-		// ReplicaSet and/or a Deployment
-		if _, err := hex.DecodeString(component); err == nil {
-			significantComponent = true
-		}
-		// Indicates we found a Pod that comes from a StatefulSet
-		if _, err := strconv.Atoi(component); err == nil {
-			significantComponent = true
-		}
-
-		if significantComponent {
-			nameComponents = nameComponents[:i+1]
-			break
-		}
+	// if the pods ends in an ordinal we can be pretty sure it's
+	// coming from a StatefulSet.
+	statefulSetRegex := regexp.MustCompile("(.*)-([0-9]+)$")
+	// ReplicasSets, and therefore Deployments because they create
+	// ReplicaSets, have a hash followed by a 5 character identity
+	// string attached to the end.
+	replicaSetRegex := regexp.MustCompile("(.*)-([0-9a-f]+)-([0-9a-z]{5})$")
+	sigNames := []string{}
+	switch {
+	case statefulSetRegex.MatchString(podName):
+		match := statefulSetRegex.FindStringSubmatch(podName)
+		appName := match[1]
+		// Add the pod name with and without the ordinal
+		sigNames = append(sigNames, podName, appName)
+	case replicaSetRegex.MatchString(podName):
+		match := replicaSetRegex.FindStringSubmatch(podName)
+		appName := match[1]
+		rsName := fmt.Sprintf("%s-%s", appName, match[2])
+		// add the app name with and without generated ReplicaSet hash
+		sigNames = append(sigNames, rsName, appName)
```

Comment:
should we also append podName here?

Reply:
I hadn't seen the full pod name show up in any logs, but it can't hurt to be extra thorough, will do :)